### PR TITLE
Multiple keys of different datatypes

### PR DESCRIPTION
**Pull request by jmsfltchr**
## What is the goal of this PR?

We add a case where an entity has two keys, each of a different datatype

## What are the changes implemented in this PR?

- Include the unique `company-number` inside the `company-name` to ensure that the `company-name` will be unique
- Rename `hashcode` to `uniqueID` to avoid confusion with the Java `hashCode`

### CODE REVIEW

File: agents/TransactionAgent.java [hunk 1]
```diff
@@ -72,8 +74,12 @@ private void insertTransaction(Pair<Long, Double> transaction, Long sellerCompan
                                 .rel("transaction_merchandise", Graql.var("product"))
 //                                .has("currency")  // TODO Add currency
```
Comment (adammitchelldev) [inline agents/TransactionAgent.java]: Do we not have currency available to complete this attribute? If not, and completing this part requires some other larger change, it would be good to create an issue now for that other change and link to it here.

File: agents/base/Agent.java [hunk 2]
```diff
@@ -132,7 +132,7 @@ public void close() {
      * @param iterationScopeId An id that uniquely identifies a concept within the scope of the agent at a particular iteration
      * @return
      */
-    protected int hashcode(int iterationScopeId) {
+    protected int uniqueID(int iterationScopeId) {
```
Comment (adammitchelldev) [inline agents/base/Agent.java]: Whilst ID is the correct abbreviation in English, we tend to follow strict camelCase rules and stylise it as `Id` instead. The reasoning is that camel case is designed to delimit word boundaries, so `XmlIdWriter` is better than `XMLIDWriter`.

tl;dr; change this method name to `uniqueId`.

File: agents/test/QueryCountE2E.java [hunk 3]
```diff
@@ -128,15 +128,20 @@ public void testTransactionAgentInsertsTheExpectedNumberOfTransactions() {
...
-                                .has("product-quantity", Graql.var("quantity"))
-//                        Graql.var("locates").isa("locates") // TODO Necessary?
+                Graql.var("country").isa("country")
+                        .has("location-name", Graql.var("country-name")),
+                Graql.var("transaction")
+                        .isa("transaction")
+                        .rel("transaction_vendor", Graql.var("c-seller"))
+                        .rel("transaction_buyer", Graql.var("c-buyer"))
+                        .rel("transaction_merchandise", Graql.var("product"))
+//                        .has("currency", Graql.var("currency-name")) // Requires reasoning over lots of results, test times out
```
Comment (adammitchelldev) [inline agents/test/QueryCountE2E.java]: Is this an issue? Should we create an issue and link it in a TODO comment?

File: agents/ProductAgent.java [hunk 4]
```diff
@@ -25,7 +25,7 @@ private void insertProduct(int iterationScopeId) {
         ).insert(
                 Graql.var("product")
                         .isa("product")
-                        .has("product-barcode", Double.valueOf(hashcode(iterationScopeId)))
+                        .has("product-barcode", Double.valueOf(uniqueID(iterationScopeId)))
```
Comment (adammitchelldev) [inline agents/ProductAgent.java]: Why is `product-barcode` a double?

Comment (jmsfltchr) [reply agents/ProductAgent.java]: This is solely because we need coverage of all datatypes as keys! If I encounter a better example as I go then I'll put it there

**Comment by jmsfltchr** [comment]
Closes #32
